README.md

```diff
@@ -1,149 +1,181 @@
 # Deferred
-
-This is an implementation of [OCaml's Deferred](https://ocaml.janestreet.com/ocaml-core/111.25.00/doc/async_kernel/#Deferred) for Swift.
-
-## Overview
-
-`Deferred` is designed for supporting asynchronous programming. An instance of
-`Deferred` represents a value that will be available at some point in the
-future. Deferred objects can trivially replace completion blocks (see
-[Running Closures Upon Fulfillment](#upon)), but also enable some higher level,
-powerful composition techniques.
-
-All properties and methods on an instance of `Deferred` can safely be called from
-multiple threads simultaneously; a lock is used internally for synchronization.
-Obviously this does not guarantee thread-safety of the contained result (which
-`Deferred` knows nothing about).
-
-An instance of `Deferred` can only be filled once. It is a programmer error to
-fill an already-filled `Deferred`, and this will result in a runtime trap. (The
-method `fillIfUnfilled` is available for conditional filling.)
-
-## Usage - Producer
+## Vital Statistics
+- What: Lets you work with values that haven't been determined yet,
+        like an array that's coming (one day!) from a web service call.
+- Who: John Gallagher <jgallagher@bignerdranch.com> wrote this.
+- Swift: v1.2 for now.
+- License: MIT
+- Carthage: Yup.
+- CocoaPods: Not yet, but feel free to submit a PR!
+
+## Intuition
+A `Deferred<Value>` is a value that might be unknown now
+but is expected to resolve to a definite `Value` at some time in the future.
+It is resolved by being "filled" with a `Value`.
+
+A `Deferred<Value>` represents a `Value`.
+If you just want to work with the eventual `Value`, use `upon`.
+If you want to produce a `Deferred<OtherValue>` from the `Deferred<Value>`,
```

Comment:
I only understood this sentence because I already have some understanding of Deferred, map, and bind. The previous sentence was a little clearer to guess at even if the reader weren't familiar with `upon` - use `upon` to have your code take action whenever the deferred gets its value. 

This one is more like `continue to chain other async work when the value comes back`. Which - is not a useful description, unlike yours. I almost didn't even comment here as I'm not sure if there is a better way to explain it concisely here, before the excellent examples below.